crates/matrix-sdk/src/notification_settings/mod.rs

```diff
@@ -12,3 +26,722 @@ pub enum RoomNotificationMode {
     /// Do not receive any notifications.
     Mute,
 }
+
+/// A high-level API to manage the client owner's push notification settings.
+#[derive(Debug, Clone)]
+pub struct NotificationSettings {
+    /// The underlying HTTP client.
+    client: Client,
+    /// Owner's account push rules. They will be updated on sync.
+    rules: Arc<RwLock<Rules>>,
+    /// Event handler for push rules event
+    push_rules_event_handler: EventHandlerHandle,
+}
+
+impl Drop for NotificationSettings {
+    fn drop(&mut self) {
+        self.client.remove_event_handler(self.push_rules_event_handler.clone());
```

Comment:
Out of curiosity, is the `clone()` required here?

Reply:
Yes, otherwise we have this error:
`move occurs because `self.push_rules_event_handler` has type `EventHandlerHandle`, which does not implement the `Copy` trait`